src/servers/Server_Zone/Inventory/Inventory.cpp

```diff
@@ -439,6 +439,30 @@ bool Core::Inventory::removeCrystal( CrystalType type, uint32_t amount )
    return true;
 }
 
+bool Core::Inventory::isOneHandedWeapon( ItemCategory weaponCategory )
+{
+   switch ( weaponCategory )
+   {
+      case ItemCategory::AlcPri:
+      case ItemCategory::ArmPri:
+      case ItemCategory::BotPri:
+      case ItemCategory::ClnPri:
+      case ItemCategory::CnjWep:
+      case ItemCategory::CrpPri:
+      case ItemCategory::FshPri:
+      case ItemCategory::GlaWep:
+      case ItemCategory::GldPri: // Goldsmith
+      case ItemCategory::LtwPri:
+      case ItemCategory::MinPri:
+      case ItemCategory::ThmWep:
+      case ItemCategory::WvrPri:
+      case ItemCategory::BlmPri: // Blacksmith
```

Comment:
iirc in game it's BSM for blacksmith

Reply:
Plenty of the values in the enum are mismatched. I'm waiting on the enum autogeneration however - since this ItemCategory doesn't actually reflect ItemUICategory.